xarray/core/accessor_str.py

```diff
@@ -883,6 +1105,42 @@ def count(
         Returns
         -------
         counts : array of int
+
+        Examples
+        --------
+        >>> da = xr.DataArray(["jjklmn", "opjjqrs", "t-euuJJvwx"], dims="x")
+        >>> da
+        <xarray.DataArray (x: 3)>
+        array(['jjklmn', 'opjjqrs', 't-euuJJvwx'], dtype='<U10')
+        Dimensions without coordinates: x
+
+        Using a string:
+        >>> da.str.count("jj")
+        <xarray.DataArray (x: 3)>
+        array([1, 1, 0])
+        Dimensions without coordinates: x
+
+        Enable case-insensitive matching by setting case to false:
+        >>> import re
+        >>> counts = da.str.count("jj", case=False)
+        >>> counts
+        <xarray.DataArray (x: 3)>
+        array([1, 1, 1])
+        Dimensions without coordinates: x
+
+        Using regex:
+        >>> counts = da.str.count(r"jj")
+        >>> counts
+        <xarray.DataArray (x: 3)>
+        array([1, 1, 0])
+        Dimensions without coordinates: x
+
+        Using a list of strings (returns a count of matches for each element):
+        >>> counts = da.str.count(["jj"])
+        >>> counts
+        <xarray.DataArray (x: 3)>
+        array([2, 2, 0])
+        Dimensions without coordinates: x
```

Comment:
Maybe we could add an example with `pat = xr.DataArray(["jj", "JJ"], dims="y")`
At least the docstring claims that this works, haven't actually tried it.

Reply:
I've tried it, and it works. I'll add it!